coral-schema/src/main/java/com/linkedin/coral/schema/avro/ViewToAvroSchemaConverter.java

```diff
@@ -84,12 +84,58 @@ public Schema toAvroSchema(String dbName, String tableOrViewName, boolean strict
    * @return avro schema for a given Dali view [dbName, viewName]
    */
   public Schema toAvroSchema(String dbName, String tableOrViewName) {
-    Preconditions.checkNotNull(dbName);
-    Preconditions.checkNotNull(tableOrViewName);
+    return toAvroSchema(dbName, tableOrViewName, false);
+  }
 
-    Schema avroSchema = inferAvroSchema(dbName, tableOrViewName, false);
+  /**
+   * This is the main API to generate the avro schema String for a given Dali view
+   *
+   * @param dbName
+   * @param tableOrViewName
+   * @param strictMode if strictMode is set to True, we will not fall back to Hive schema if avro.schema.literal
+   *                        is missing in table properties. In addition, original namespace in base tables will be preserved.
+   *                   if strictMode is set to False, we will fall back to Hive schema if avro.schema.literal
+   *                        is missing in table properties. A new set of namespace will be generated for
+   *                        the resulting schema. The rule is as follows:
+   *                                1. Top level namespace is dbName.tableOrViewName
+   *                                2. Nested namespace is parentNamespace.parentFieldName
+   * @param isPrettyPrint if pretty print string is required
+   *
+   * @return avro schema String for a given Dali view [dbName, viewName]
+   */
+  public String toAvroSchemaString(String dbName, String tableOrViewName, boolean strictMode, boolean isPrettyPrint) {
```

Comment:
Should we remove this API and return the pretty print by default? What your thoughts on this @wmoustafa ?

Reply:
I think pretty print will not compress the information. It may not have a better performance for extra large schema.
But, somehow it is very useful on debugging purpose.

Reply:
Do you think it is cleaner to remove isPrettyPrint option? cc: @wmoustafa